warehouse/redshift.go

```diff
@@ -295,3 +325,60 @@ func (rs *Redshift) DoesTableExist(name string) bool {
 	}
 	return (exists != 0)
 }
+
+func (rs *Redshift) getTableColumns(name string) []string {
+	ctx := context.Background()
+	log.Printf("Fetching columns for table %s", name)
+	rows, err := rs.conn.QueryContext(ctx,
+		"SELECT \"column\" FROM pg_table_def WHERE schemaname = 'public' AND tablename = $1;", name)
+	if err != nil {
+		log.Fatal(err)
+	}
+	var columns []string
+
+	defer rows.Close()
+	for rows.Next(){
+		var column string
+		err = rows.Scan(&column)
+		columns = append(columns, column)
+	}
+	err = rows.Err() // get any error encountered during iteration
+	if err != nil {
+		log.Fatal(err)
+	}
+	return columns
+}
+
+func (rs *Redshift) getAlterTableStatements(fields []WarehouseField, tableName string) []string {
+	var alterTableStmts []string
+	for _, f := range fields {
+		alterTableStmts = append(alterTableStmts, fmt.Sprintf("ALTER TABLE %s ADD COLUMN %s %s;", tableName, f.Name, f.DBType))
+	}
+
+	return alterTableStmts
+}
+
+func (rs *Redshift) getMissingFields(schema Schema, tableColumns []string) []WarehouseField {
+	existingColumns := make(map[string]bool)
```

Comment:
Some people prefer to make a set in go with `map[string]struct{}`, since `struct{}` takes no storage space. But others think that syntax is too unwieldy. I can see either argument, just wanted to make sure you knew about that subtlety.

Reply:
I would do just a one-off for now, and then look at refactoring when we do the BigQuery implementation, since we expect there to be some significant refactoring there anyway.